mlir/lib/Analysis/Presburger/Coalesce.cpp

```diff
@@ -0,0 +1,686 @@
+#include "mlir/Analysis/Presburger/Coalesce.h"
+#include "mlir/Analysis/AffineStructures.h"
+#include "mlir/Analysis/Presburger/Simplex.h"
+#include "mlir/Analysis/Presburger/Set.h"
+#include <iostream>
+
+using namespace mlir;
+// struct for classified constraints
+struct Info {
+  SmallVector<SmallVector<int64_t, 8>, 8> redundant;
+  SmallVector<SmallVector<int64_t, 8>, 8> cut;
+  Optional<SmallVector<int64_t, 8>> adj_ineq;
+  Optional<SmallVector<int64_t, 8>> t;
+};
+
+void dumpInfo(Info *info);
```

Comment:
Wouldn't this enlarge the structs themselves by a pointer value? As dumping a struct isn't actually important for the algorithms, this would just make us lose performance for no reason.

Reply:
As far as I know this shouldn't be the case as the method isn't virtual. Therefore the function to call is known at compile time. If I understand C++ compilation correctly, this wouldn't make a performance difference.

Reply:
done